site/cypress/e2e/cypress-feature/late_days_cache.spec.js

```diff
@@ -0,0 +1,666 @@
+                        .contains(g_title)
+                        .siblings('td[data-before-content="Days Submitted Late"]')
+                        .should('not.have.value', 'N/A').then((value) => {
+                            // Grab the # of days late (should be > 0)
+                            const days_late = parseInt(value.text());
+                            expect(days_late).to.be.greaterThan(0);
+                            all_late_users[g_id][user_id] = days_late;
+
+                            // Wait for login change to take place
+                            cy.wait(3000);
+                            // Generate predicted status
+                            const status = predictedStatus(days_allowed, days_late, 0);
+
+                            // Find late day status within the row in the late day usage table
+                            cy.get('td[data-before-content="Event/Assignment"]')
+                                .contains(g_title)
+                                .siblings('td[data-before-content="Status"]')
+                                .contains(status)
+                                .should('exist');
+                        });
+                    cy.logout();
+                }
+            });
+
+            it('should have 0 late days used on bulk late days table', () => {
+                cy.visit(['sample', 'bulk_late_days']);
+                cy.login('instructor');
+
+                // All statuses should be bad, 0 late days should be charged
+                for (const user_id in all_late_users[g_id]) {
+                    cy.get(`[data-user="${user_id}"] > [data-before-content="${g_title}"]`)
+                        .contains('0')
+                        .should('exist');
+                }
+            });
+        });
+    }
+
+    describe('Test changes to late days allowed table', () => {
+        const late_days_remaining = {};
+        it('should grant students with 2 late days', () => {
+            cy.visit(['sample', 'late_days']);
+            cy.login('instructor');
+
+            for (const user_id of all_user_ids) {
+                const days = Math.floor(Math.random() * 2) + 1;
+                // update the number of late days
+                cy.get('#user_id').type(user_id);
+                cy.get('#datestamp').type('1972-01-01', {force: true});
+                cy.get('#user_id').click(); // dismiss the calendar view
+                cy.get('#late_days').clear();
+                cy.get('#late_days').type(days);
+                cy.get('input[type=submit]').click();
+                cy.wait(3000);
+                late_days_remaining[user_id] = days;
+            }
+        });
+
+        it('should make bulk late days has been emptied out', () => {
+            cy.visit(['sample', 'bulk_late_days']);
+            cy.login('instructor');
+
+            for (const test_case of test_info) {
+                const { g_id, g_title } = test_case;
```

Comment:
I am partial to longer variable names, as it makes the code easier to read, but that's just personal preference, and not a problem for merging.